### PR TITLE
Issue/552/improve 2h functions

### LINKED ISSUE #552: Improve the 2 halo functions
Current 2h functions (`eval_surface_density_2h`, `eval_excess_surface_density_2h` and their `_*` counterparts), have an inefficient integration scheme and also hardcoded values for the integrals. Should be improved.

### PR DESCRIPTION
**Pull request by rancesol**
Closes #552 
- Improve efficiency of the `*_2h` functions by replacing `interp1d` with `splev` and `splrep`.
- Changed previously hard coded values in the `*_2h` functions into optional user defined arguments.
- Added shape requirement to the validate_argument function in clmm/utils.py.

### CODE REVIEW

**Comment by m-aguena** [comment]
@rancesol the changes look good, it's just missing a unit test for the new functionality you added to validate_argument. Can you add it to `tests/test_utils.py`? Thanks

**Review by m-aguena — CHANGES_REQUESTED** [review]
https://github.com/LSSTDESC/CLMM/pull/575#issuecomment-1474180580

File: tests/test_utils.py [hunk 1]
```diff
@@ -425,6 +425,12 @@ def test_validate_argument():
         assert validate_argument(loc, argname, ('float_array', str), argmin=0, argmax=4,
                                  eqmin=True, eqmax=True) is None
 
+    # Test shape
+    for argname in ('int', 'float', 'str') :
+        assert validate_argument(loc, argname, (str, 'float_array'), shape=()) is None
+    for argname in ('int_array', 'float_array') :
+        assert validate_argument(loc, argname, 'float_array', shape=(2,)) is None
+
     assert_raises(TypeError, validate_argument, loc, 'str', ('float_array', str), argmin=0)
```
Comment (m-aguena) [inline tests/test_utils.py:435]: Hi @rancesol , what you also want here is to make sure the `validate_argument` raises an error when the shape is wrong. Try using a `assert_raises` test

**Review by m-aguena — CHANGES_REQUESTED** [review]

**Review by marina-ricci — APPROVED** [review]
Thank you @rancesol for these changes! It looks good to me.

**Pull request merged**